src/App.jsx

```diff
@@ -13,34 +13,83 @@
   'Jam',
   'Garlic',
 ];
+const SORT_FIELD_ALPHABETICCALY = 'alphabetically';
+const SORT_FIELD_LENGTH = 'length';
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+function getPreparedGoods(goods, sortField, isReversed) {
+  const preparedGoods = [...goods];
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+  preparedGoods.sort((good1, good2) => {
+    switch (sortField) {
+      case SORT_FIELD_ALPHABETICCALY:
+        return good1.localeCompare(good2);
+      case SORT_FIELD_LENGTH:
+        return good1.length - good2.length;
+      default:
+        return 0;
+    }
+  });
+  if (isReversed) {
+    preparedGoods.reverse();
+  }
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+  return preparedGoods;
+}
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+export const App = () => {
+  const [sortField, setSortField] = useState('');
+  const [isReversed, setIsReversed] = useState(false);
+  const visibleGoods = getPreparedGoods(goodsFromServer, sortField, isReversed);
+  const isModified = sortField !== '' || isReversed;
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          onClick={() => setSortField(SORT_FIELD_ALPHABETICCALY)}
```

Comment:
Ensure that the corrected constant name `SORT_FIELD_ALPHABETICALLY` is used here as well.